syntax = "proto3";

package itinerary;

service ItineraryService {
    rpc GetTravelPlans (Token) returns (TravelPlanResponse);
}

message TravelPlan {
  string id = 1;
  string title = 2;
  string description = 3;
  optional string location_name = 4;
  optional double location_lat = 5;
  optional double location_long = 6;
  optional int64 arrival_date = 7;
  optional int64 departure_date = 8;
  string user_email = 9;
}

message TravelPlanResponse {
  repeated TravelPlan data = 1;
}

message Token {
  string token = 1;
}
